### PR TITLE
Add the ability to give flexible children loose constraints

### PR DESCRIPTION
**Pull request by abarth**
There are some use cases where we want to add some content to the side of
another widget and have that widget able to expand to fill the remaining space.

Related to #3633

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/rendering/flex.dart [hunk 1]
```diff
@@ -7,6 +7,15 @@ import 'dart:math' as math;
 import 'box.dart';
 import 'object.dart';
 
+/// Whether to apply tight or loose constraints to flexible children.
+enum FlexConstraintType {
+  /// Require the child to fill the flexible space.
+  tight,
+
+  /// Permit the child to use less than all the flexible space.
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/flex.dart]: mention that the space becomes "free" space and isn't allocated to the other children if all the space isn't used

**Comment by Hixie** [comment]
LGTM

**Comment by abarth** [comment]
I'm going to try another approach before landing this patch.

**Comment by krisgiesing** [comment]
I'm a little confused... if children are currently required to be flush inside the container, then what does mainAxisAlignment do?

**Comment by abarth** [comment]
Yeah, I think this patch makes the system too confusing.

Without this patch, the answer to your question is straightforward.  If there are no flexible children, the inflexible children might not fill the available horizontal space, which is why there's an option for what to do with the remaining free space.

**Pull request closed without merging**